ovn_cluster.sh

```diff
@@ -592,95 +706,106 @@ function start() {
+ovn-nbctl lsp-set-type sw1\$az-lr\$az router
+ovn-nbctl lsp-set-addresses sw1\$az-lr\$az router
+ovn-nbctl lsp-set-options sw1\$az-lr\$az router-port=lr\$az-sw1\$az
+
+ovn-nbctl ls-add public\$az
+ovn-nbctl lrp-add lr\$az lr\$az-public\$az 00:0\$az:20:20:12:13 172.16.\$az.100/24 300\$az::a/64
+ovn-nbctl lsp-add public\$az public\$az-lr\$az
+ovn-nbctl lsp-set-type public\$az-lr\$az router
+ovn-nbctl lsp-set-addresses public\$az-lr\$az router
+ovn-nbctl lsp-set-options public\$az-lr\$az router-port=lr\$az-public\$az
 
 # localnet port
-ovn-nbctl lsp-add public ln-public
-ovn-nbctl lsp-set-type ln-public localnet
-ovn-nbctl lsp-set-addresses ln-public unknown
-ovn-nbctl lsp-set-options ln-public network_name=public
+ovn-nbctl lsp-add public\$az ln-public\$az
+ovn-nbctl lsp-set-type ln-public\$az localnet
+ovn-nbctl lsp-set-addresses ln-public\$az unknown
+ovn-nbctl lsp-set-options ln-public\$az network_name=public
 
 # schedule the gw router port to a chassis.
-ovn-nbctl lrp-set-gateway-chassis lr0-public ovn-gw-1 20
+ovn-nbctl lrp-set-gateway-chassis lr\$az-public\$az ovn-gw-\$az 20
 
 # Create NAT entries for the ports
 
-# sw0-port1
-ovn-nbctl lr-nat-add lr0 dnat_and_snat 172.16.0.110 10.0.0.3 sw0-port1 30:54:00:00:00:03
-ovn-nbctl lr-nat-add lr0 dnat_and_snat 3000::c 1000::3 sw0-port1 40:54:00:00:00:03
-# sw1-port1
-ovn-nbctl lr-nat-add lr0 dnat_and_snat 172.16.0.120 20.0.0.3 sw1-port1 30:54:00:00:00:04
-ovn-nbctl lr-nat-add lr0 dnat_and_snat 3000::d 2000::3 sw1-port1 40:54:00:00:00:04
+# sw0\$az-port1
+ovn-nbctl lr-nat-add lr\$az dnat_and_snat 172.16.\$az.110 1\$az.0.0.3 sw0\$az-port1 30:5\$az:00:00:00:03
+ovn-nbctl lr-nat-add lr\$az dnat_and_snat 300\$az::c 100\$az::3 sw0\$az-port1 40:5\$az:00:00:00:03
+# sw1\$az-port1
+ovn-nbctl lr-nat-add lr\$az dnat_and_snat 172.16.\$az.120 2\$az.0.0.3 sw1\$az-port1 30:5\$az:00:00:00:04
+ovn-nbctl lr-nat-add lr\$az dnat_and_snat 300\$az::d 200\$az::3 sw1\$az-port1 40:5\$az:00:00:00:04
 
 # Add a snat entry
-ovn-nbctl lr-nat-add lr0 snat 172.16.0.100 10.0.0.0/24
-ovn-nbctl lr-nat-add lr0 snat 172.16.0.100 20.0.0.0/24
+ovn-nbctl lr-nat-add lr\$az snat 172.16.\$az.100 1\$az.0.0.0/24
+ovn-nbctl lr-nat-add lr\$az snat 172.16.\$az.100 2\$az.0.0.0/24
+
+if [ "\$ic" == "yes" ]; then
+    ovn-nbctl lrp-add lr\$az lr\$az-ts1 aa:aa:aa:aa:aa:0\$az 5.0.0.\$az/24
+    ovn-nbctl lsp-add ts1 ts1-lr\$az -- lsp-set-addresses ts1-lr\$az -- lsp-set-type ts1-lr\$az router -- lsp-set-options ts1-lr\$az  router-port=lr\$az-ts1
+    ovn-nbctl lrp-set-gateway-chassis lr\$az-ts1 ovn-gw-\$az 1
+fi
 
 EOF
     chmod 0755 ${FAKENODE_MNT_DIR}/create_ovn_res.sh
     if [ "$OVN_DB_CLUSTER" = "yes" ]; then
-        central=${CENTRAL_NAME}-1
+        central=${CENTRAL_PREFIX}${az}-1
     else
-        central=${CENTRAL_NAME}
+        central=${CENTRAL_PREFIX}${az}
     fi
-    ${RUNC_CMD} exec ${central} bash /data/create_ovn_res.sh
+    ${RUNC_CMD} exec ${central} bash /data/create_ovn_res.sh $az $ic
```

Comment:
If we unify the naming scheme for the first DB container this can be written as a single line:
```
${RUNC_CMD} exec ${CENTRAL_NAME} bash /data/create_ovn_res.sh $az $ic
```

Reply:
@dceara ack, I will fix it

Reply:
I think that we can improve this.  There's no need for a separate CENTRAL_NAME variable, we can just always create an array of CENTRAL_NAMES (sometimes it might have a singe entry).  Then we can set CENTRAL_NAME to be CENTRAL_NAMES[0].

While at it I think we should also refactor all the name setting, we do the same for loops a bunch of times in the script.

What do you think of the following incremental changes?
- refactor the setting the node names arrays:
https://github.com/dceara/ovn-fake-multinode/commit/00db261886a207a6caf1d23040e535a5644464da
- unify the way we generate central names:
https://github.com/dceara/ovn-fake-multinode/commit/4780405837b6c4d3b983e8342167ce96f1abb0e5

Reply:
@dceara can we avoid changing the name scheme here and use `ovn-central-az1-1` when we have just one central node?

Reply:
The refactor is OK, the problem is the second commit (https://github.com/dceara/ovn-fake-multinode/commit/4780405837b6c4d3b983e8342167ce96f1abb0e5), let's skip that one.

However, in the current form of the PR when I try this with ovn-heater ([a modified version of your IC PR](https://github.com/dceara/ovn-heater-1/tree/ovn-ic-review)) with n_az set to 2 I get:
```
CONTAINER ID  IMAGE                                COMMAND         CREATED             STATUS             PORTS       NAMES
c1d5a3728d2b  localhost/ovn/ovn-tester:latest      /usr/sbin/init  About a minute ago  Up About a minute              ovn-tester
dfdbe701c7f4  localhost/ovn/ovn-multi-node:latest  /usr/sbin/init  About a minute ago  Up About a minute              ovn-central-az01-1
22de612db4b1  localhost/ovn/ovn-multi-node:latest  /usr/sbin/init  About a minute ago  Up About a minute              ovn-central-az01-relay-1
441ec5947ea9  localhost/ovn/ovn-multi-node:latest  /usr/sbin/init  About a minute ago  Up About a minute              ovn-central-az01-relay-2
eebb944de084  localhost/ovn/ovn-multi-node:latest  /usr/sbin/init  About a minute ago  Up About a minute              ovn-central-az01-relay-3
0ac91f0ca9a4  localhost/ovn/ovn-multi-node:latest  /usr/sbin/init  About a minute ago  Up About a minute              ovn-central-az11-1
4f235bba16b3  localhost/ovn/ovn-multi-node:latest  /usr/sbin/init  About a minute ago  Up About a minute              ovn-central-az11-relay-1
09e45d89ae57  localhost/ovn/ovn-multi-node:latest  /usr/sbin/init  About a minute ago  Up About a minute              ovn-central-az11-relay-2
b24d1b219d23  localhost/ovn/ovn-multi-node:latest  /usr/sbin/init  About a minute ago  Up About a minute              ovn-central-az11-relay-3
```

Can we find a way to get rid of that annoying extra 1?

Reply:
@dceara applying the [refactor patch](https://github.com/dceara/ovn-fake-multinode/commit/4780405837b6c4d3b983e8342167ce96f1abb0e5) this is what I get:
`
CONTAINER ID  IMAGE                                NAMES
45184e8ecd06  localhost/ovn/ovn-multi-node:latest  ovn-central-az1-
62638049c14a  localhost/ovn/ovn-multi-node:latest  ovn-central-az1-
1e9e2b189a0f  localhost/ovn/ovn-multi-node:latest  ovn-central-az1-
b8a1b219adbd  localhost/ovn/ovn-multi-node:latest  ovn-central-az2-
68f90b8a7e28  localhost/ovn/ovn-multi-node:latest  ovn-central-az2-
8905884a5deb  localhost/ovn/ovn-multi-node:latest  ovn-central-az2-
69ad915b73fe  localhost/ovn/ovn-multi-node:latest  ovn-gw-1
560712dc1582  localhost/ovn/ovn-multi-node:latest  ovn-gw-2
c9bb9f211323  localhost/ovn/ovn-multi-node:latest  ovn-central-az1-
6da8d434bda1  localhost/ovn/ovn-multi-node:latest  ovn-central-az2-
e1d68de523dc  localhost/ovn/ovn-multi-node:latest  ovn-chassis-1
11021c00379d  localhost/ovn/ovn-multi-node:latest  ovn-chassis-2
b8a004be61bf  localhost/ovn/ovn-multi-node:latest  ovn-chassis-3
b15efd403bee  localhost/ovn/ovn-multi-node:latest  ovn-chassis-4
`

Reply:
Yes, that's what I was hoping for.  However, I'm OK to anything that works without too much hassle.

What ovn-heater needs for IC (as far as I can tell) is a way to start a new central node (or nodes if clustered dbs are used) - so a new AZ central component - in one individual ovn_cluster.sh invocation.

In my test I was doing:
```
  - name: Start central containers
    environment:
      CENTRAL_COUNT: 1
      CENTRAL_PREFIX: 'ovn-central-az{{ item|int }}'
      CHASSIS_COUNT: 0
      CREATE_FAKE_VMS: no
      ENABLE_SSL:  '{{ enable_ssl }}'
      GW_COUNT: 0
      IP_CIDR:  "{{ node_net|ansible.utils.ipaddr('prefix') }}"
      IP_HOST:  "{{ node_net|ansible.utils.ipaddr('network') }}"
      IP_START: "{{ node_net|ansible.utils.ipmath(2 + item|int * n_ips|int) }}"
      OVN_START_IC_DBS: '{{ (item|int == 0) | ansible.builtin.ternary("yes", "no") }}'
      CENTRAL_IC_ID:    '{{ central_ic_id }}'
      CENTRAL_IC_IP:    '{{ node_net|ansible.utils.ipmath(2) }}'
      OVN_DB_CLUSTER:   '{{ clustered_db }}'
      OVN_MONITOR_ALL:  '{{ monitor_all }}'
      RELAY_COUNT:      '{{ n_relays }}'
    ansible.builtin.shell: |
      ./ovn_cluster.sh start
    args:
      chdir: "{{ ovn_fake_multinode_target_path }}/ovn-fake-multinode"
    loop: "{{ range(groups['central_hosts'].index(inventory_hostname),
                    n_az|int, groups['central_hosts']|length) | list }}"
```

But if we could explicitly pass CENTRAL_NAME instead and get a working cluster that's even better.

Reply:
@dceara ack, I will add the capability to specify CENTRAL_NAME in ovn-fake-multinode